index.js

```diff
@@ -31,7 +31,10 @@ module.exports = (input, opts) => Promise.resolve().then(() => {
 	const stat = pify(opts.fs.stat);
 
 	const make = pth => {
-		return mkdir(pth, opts.mode)
+		return mkdir(pth, {
```

Comment:
This is a problem because someone could set opts.fs containing an alternate implementation mkdir.

I think you can only use mkdir(pth, options_object) only when node.js is at least 10.12 and opts.fs.mkdir === fs.mkdir.  In all other cases the legacy mkdir(pth, opts.mode) method needs to be used.

The same comment applies to the `sync` method, it needs to check the version and verify opts.fs.mkdirSync === fs.mkdirSync before using the options object.